### PR TITLE
[SPARK-50035][SS] Add support for explicit handleExpiredTimer function part of the stateful processor

### PR DESCRIPTION
**Pull request by anishshri-db**
### What changes were proposed in this pull request?
Add support for explicit handleExpiredTimer function part of the stateful processor

### Why are the changes needed?
Separate function will provide for cleaner UX and eliminates need for special handling of empty input rows and expired timer validity

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
Existing unit tests

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by anishshri-db** [comment]
cc - @ericm-db @HeartSaVioR - PTAL, thx !

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ExpiredTimerInfoImpl.scala [hunk 1]
```diff
@@ -16,21 +16,15 @@
...
  * Class that provides a concrete implementation that can be used to provide access to expired
  * timer's expiry time. These values are only relevant if the ExpiredTimerInfo
  * is valid.
- * @param isValid - boolean to check if the provided ExpiredTimerInfo is valid
  * @param expiryTimeInMsOpt - option to expired timer's expiry time as milliseconds in epoch time
  */
-class ExpiredTimerInfoImpl(
-    isValid: Boolean,
-    expiryTimeInMsOpt: Option[Long] = None,
-    timeMode: TimeMode = TimeMode.None()) extends ExpiredTimerInfo {
```
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ExpiredTimerInfoImpl.scala:31]: Is it a side effect? It's obvious to remove isValid, but timeMode does not seem to be obvious.

**Review by HeartSaVioR — COMMENTED** [review]
Mostly looks good. Just need one clarification about additional change than the PR title.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ExpiredTimerInfoImpl.scala (hunk 1 above)
Comment (anishshri-db) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ExpiredTimerInfoImpl.scala:31]: Yea - afaict - i don't think its used anywhere. Not sure why we had it before. So just removed it as part of cleanup

**Review by HeartSaVioR — APPROVED** [review]
+1 pending CI

**Pull request closed without merging**